lib/ansible/parsing/yaml/utils.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2018 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.six import text_type
+
+
+def convert_yaml_objects_to_native(obj):
```

Comment:
There are other ways to handle this all in pyyaml as well if we want a truly generic thing, but assuming we have a priori knowledge of the actual types or their base classes, this is easy to handle by adding custom representers or multi_representers:
```
import yaml
from yaml.representer import SafeRepresenter
class MyStr(str):
    pass
SafeRepresenter.add_representer(MyStr, lambda obj, value: SafeRepresenter.represent_str(obj, str(value)))
x = {
	'foo': MyStr('bar'),
	'baz': [1,2,3]
}
print(yaml.dump(x, Dumper=yaml.CSafeDumper))
```